GoMeet/src/main/webapp/js/email-dao.js

```diff
@@ -0,0 +1,38 @@
+class EmailDAO {
+  /** Data Access Object for sending emails */
+
+  static endpoint = '/email'; // URL endpoint of the associated servlet
+  
+  /**
+   * Completes a POST request to send an email invitation with the 
+   * meeting event link to all addresses in the guest list 
+   * @param {String} meetingTimeId the unique entity key of the 
+   * MeetingEvent that the guest list is invited to 
+   * @param {List} guestList the list of guests' email addresses 
+   * @return JSON containing the sent status of the emails  
+   */
+  static async inviteGuestsToMeeting(meetingEventId, guestList) {
+    if (meetingEventId === null || meetingEventId === undefined ||
+        guestList === null || guestList === undefined) {
+      throw new Error(INSUFFICIENT_REQUEST_PARAM);
+    }
+
+    if (typeof meetingEventId !== 'string' || typeof guestList !== 'object') {
```

Comment:
Hmm maybe you should check whether guestList is an Array (you can do it like [this](https://stackoverflow.com/questions/4775722/how-to-check-if-an-object-is-an-array))?

Reply:
Good point